Core.ElasticSearch/Projections/ElasticSearchProjection.cs

```diff
@@ -30,7 +30,7 @@ public async Task Handle(EventEnvelope<TEvent> eventEnvelope, CancellationToken
         var entity = (await elasticClient.GetAsync<TView>(id, i => i.Index(indexName), ct).ConfigureAwait(false))?.Source ??
                      (TView) Activator.CreateInstance(typeof(TView), true)!;
 
-        entity.When(eventEnvelope);
+        entity.Evolve(eventEnvelope);
```

Comment:
if it mutates, it should be called Apply IMO - Evolve only makes sense if it's returning the evolved form